cartography/data/jobs/cleanup/aws_import_rds_instances_cleanup.json

```diff
@@ -1,8 +1,15 @@
 {
-  "statements": [{
-    "query": "MATCH (n:RDSInstance)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}}) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
-    "iterative": true,
-    "iterationsize": 100
-  }],
+  "statements": [
+    {
+      "query": "MATCH (sng:DBSubnetGroup)<-[:PART_OF_DB_SUBNET_GROUP]-(:RDSInstance)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}}) WHERE sng.lastupdated <> {UPDATE_TAG} WITH sng LIMIT {LIMIT_SIZE} DETACH DELETE (sng) return COUNT(*) as TotalCompleted",
+      "iterative": true,
+      "iterationsize": 100
+    },
+    {
+      "query": "MATCH (n:RDSInstance)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}}) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
+      "iterative": true,
+      "iterationsize": 100
+    }
+  ],
```

Comment:
missing MERGE(sng)-[r:RESOURCE]->(subnet) cleanup ?

Reply:
EC2Subnets are already cleaned up [here](https://github.com/lyft/cartography/blob/master/cartography/data/jobs/cleanup/aws_import_ec2_instances_cleanup.json#L13).

Reply:
The MERGE statement implies there are possibilities for creating a new else we could be explicit with MATCH.

It seems reasonable to keep MERGE as subnet may have been created since we did EC2 sync for example (creation while sync in running)... A general rule I try to keep for myself is if we do MERGE, we should cleanup that specific node or relationship.

Reply:
> The MERGE statement implies there are possibilities for creating a new else we could be explicit with MATCH.

Ohh thanks good point, I missed that!

> my 0.02$, sounds good to me to keep MERGE as subnet may have been created since we did EC2 sync for example (creation while sync in running)

In the specific case you mentioned, the ECSubnet wouldn't have the other relationships that it normally has.  I think I would rather not include the incomplete EC2Subnet in this case, so I like the idea of being strict with match.

Reply:
if our goal is to represent the state as accurately as possible, MERGE makes sense to me. It also avoids trying to figure out what other modules are doing and ordering. Imagine RDS runs before EC2 (lamba, queue ...) in the future...

Reply:
I'm deciding to not add new EC2Subnet nodes here.  We will just rely on the data provided by the previously run EC2 module since in this RDS module the only field we have is the subnet ID and nothing else.

Reply:
doing merge as per convo